io_scene_niftools/modules/nif_import/animation/morph.py

```diff
@@ -64,50 +64,48 @@ def import_morph_controller(self, n_node, b_obj):
             morphData = n_morphCtrl.data
             if morphData.num_morphs:
                 # get name for base key
-                keyname = morphData.morphs[0].frame_name.decode()
+                morph_data = morphData.morphs[0]
+                keyname = morph_data.frame_name.decode()
                 if not keyname:
                     keyname = 'Base'
 
-                # insert base key at frame 1, using relative keys
+                # insert base key, using relative keys
                 sk_basis = b_obj.shape_key_add(name=keyname)
 
                 # get base vectors and import all morphs
-                baseverts = morphData.morphs[0].vectors
+                baseverts = morph_data.vectors
 
-                shape_action = self.create_action(b_obj.data.shape_keys, b_obj.name + "-Morphs")
+                shape_action = self.create_action(b_obj.data.shape_keys, f"{b_obj.name}-Morphs")
                 
-                for idxMorph in range(1, morphData.num_morphs):
+                for morph_i in range(1, morphData.num_morphs):
+                    morph_data = morphData.morphs[morph_i]
```

Comment:
This shadow the outer `morph_data`

Reply:
it's ok, is just needed at the beginning for the base shape key